syz-fuzzer/fuzzer.go

```diff
@@ -514,28 +514,67 @@ func (fuzzer *Fuzzer) checkDisabledCalls(p *prog.Prog) {
 	}
 }
 
-func (fuzzer *FuzzerSnapshot) chooseProgram(r *rand.Rand) *prog.Prog {
-	randVal := r.Int63n(fuzzer.sumPrios + 1)
-	idx := sort.Search(len(fuzzer.corpusPrios), func(i int) bool {
-		return fuzzer.corpusPrios[i] >= randVal
-	})
-	return fuzzer.corpus[idx]
+func (fuzzer *Fuzzer) updateCorpusMap(hashes signal.Signal, p *prog.Prog) {
+	for hashRaw := range hashes.Serialize().Elems {
+		hash := uint32(hashRaw)
+		old, ok := fuzzer.perHashProgs[hash]
+		if !ok {
+			fuzzer.allHashes = append(fuzzer.allHashes, hash)
```

Comment:
Q: Maybe allocate more slice capacity from the very beginning? This should help waste fewer memory (re)allocations.